projects/RabbitMQ.Client/client/impl/AsyncRpcContinuations.cs

```diff
@@ -31,26 +31,70 @@
 
 using System;
 using System.Runtime.CompilerServices;
+using System.Threading;
 using System.Threading.Tasks;
 using RabbitMQ.Client.client.framing;
 using RabbitMQ.Client.Exceptions;
 using RabbitMQ.Client.Framing.Impl;
 
 namespace RabbitMQ.Client.Impl
 {
-    internal abstract class AsyncRpcContinuation<T> : IRpcContinuation
+    internal abstract class AsyncRpcContinuation<T> : IRpcContinuation, IDisposable
     {
+        private readonly CancellationTokenSource _ct;
+
         protected readonly TaskCompletionSource<T> _tcs = new TaskCompletionSource<T>(TaskCreationOptions.RunContinuationsAsynchronously);
 
+        private bool _disposedValue;
+
+        public AsyncRpcContinuation(TimeSpan continuationTimeout)
+        {
+            _ct = new CancellationTokenSource(continuationTimeout);
```

Comment:
It might be a good idea to keep a "pool" of CancellationTokenSource objects around (using a ConcurrentStack for example with a set max limit) since the CancellationTokenSource objects can be [Reset](https://learn.microsoft.com/en-us/dotnet/api/system.threading.cancellationtokensource.tryreset?view=net-7.0) (only in NET 5, 6 and 7 though) for reuse later. This is good to minimize allocation overhead.

Reply:
Since we'll be targeting `netstandard2.0` for the foreseeable future I'll add this as a future TODO.